js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSEnv.scala

```diff
@@ -9,11 +9,47 @@
 
 package org.scalajs.jsenv
 
-import org.scalajs.core.tools.io._
-import org.scalajs.core.tools.classpath._
-import org.scalajs.core.tools.logging._
+import org.scalajs.core.tools.io.VirtualJSFile
+import org.scalajs.core.tools.jsdep.ResolvedJSDependency
+import org.scalajs.core.tools.linker.LinkingUnit
 
 trait AsyncJSEnv extends JSEnv {
-  def asyncRunner(classpath: CompleteClasspath, code: VirtualJSFile,
-      logger: Logger, console: JSConsole): AsyncJSRunner
+  def asyncRunner(libs: Seq[ResolvedJSDependency], sjsCode: VirtualJSFile,
+      code: VirtualJSFile): AsyncJSRunner
+
+  override def loadLibs(libs: Seq[ResolvedJSDependency],
+      sjsCode: VirtualJSFile): AsyncLibJSEnv = {
+    new AsyncLibJSEnv {
+      def unloadLibs: AsyncJSEnv = AsyncJSEnv.this
+
+      def jsRunner(code: VirtualJSFile): JSRunner =
+        AsyncJSEnv.this.jsRunner(libs, sjsCode, code)
+
+      def asyncRunner(code: VirtualJSFile): AsyncJSRunner =
+        AsyncJSEnv.this.asyncRunner(libs, sjsCode, code)
+    }
+  }
+}
+
+trait LinkingUnitAsyncJSEnv extends LinkingUnitJSEnv with AsyncJSEnv {
+  def asyncRunner(libs: Seq[ResolvedJSDependency], linkingUnit: LinkingUnit,
+      code: VirtualJSFile): AsyncJSRunner
```

Comment:
Yes, exactly. But since it is a `JSEnv`, it is not a backend in the sense of our typing. For a backend that creates a JS file, that whole jazz would work fine.

Currently, Rhino does find the symbols it needs, since the `Linker` it uses has the `BasicLinkerBackend`.